src/pages/SearchPage/SearchPage.test.js

```diff
@@ -0,0 +1,203 @@
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(axiosMock.get).toHaveBeenCalledTimes(2);
+    expect(axiosMock.get).toBeCalledWith(expect.stringMatching(NEW_SUBREDDIT));
+  });
+  test('Search button is disabled when loading takes place', async () => {
+    const { getByTestId } = setup();
+    const button = getByTestId('search-button');
+    const input = getByTestId('search-input');
+    expect(input).toBeInTheDocument();
+    expect(input.value).toEqual(DEFAULT_SUBREDDIT);
+    expect(button).toBeDisabled();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Error and heatmap display states', () => {
+  test('Will show an error message when there are no results', async () => {
+    const { getByTestId } = setup();
+    expect(getByTestId('loading-spinner')).toBeInTheDocument();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    const error = getByTestId('error');
+    expect(error).toBeInTheDocument();
+    expect(error.textContent).toBe('0 results returned.');
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+  test('Will show an error message when the API call fails', async () => {
+    const { getByTestId } = setup('fail');
+    expect(getByTestId('loading-spinner')).toBeInTheDocument();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    const error = getByTestId('error');
+    expect(error).toBeInTheDocument();
+    expect(error.textContent).toBe('There was an error processing your request.');
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+  test('Will load heatmap with real test data', async () => {
+    const { getByTestId } = setup('pass', dummyPosts);
+    expect(getByTestId('loading-spinner')).toBeInTheDocument();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(getByTestId('heatmap')).toBeInTheDocument();
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Heatmap values', () => {
+  test('Heatmap values matches dummy data', async () => {
+    const dummyHeatmapValues = getHeatmapValues(dummyPosts);
+    const { getByTestId, getAllByTestId } = setup('pass', dummyPosts);
+    expect(getByTestId('loading-spinner')).toBeInTheDocument();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    const heatmap = getByTestId('heatmap');
+    expect(heatmap).toBeInTheDocument();
+    // Checking buttons
+    const heatmapValues = getAllByTestId('heatmap-button').map((button) => parseInt(button.textContent, 10));
+    // Make sure number of buttons in heatmap is correct
+    expect(heatmapValues.length).toEqual(NUM_GRID_ITEMS);
+    // Heatmap values must match dummy data values
+    expect(heatmapValues).toEqual(dummyHeatmapValues);
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+  test('Heatmap elements use the correct theme colours', async () => {
```

Comment:
Even nicer! 

Fyi: this might also be tested with a visual regression test with cypress, percy etc. Those basically take a screenshot and compare them to a previous version. But that's out of scope for this project IMO. Just good to know for you when you have interviews or get a job. Not a lot of companies have those and they can be valuable and easy to implement